test/broadcast.jl

```diff
@@ -951,6 +951,46 @@ p0 = copy(p)
 @test repr(.!) == "Base.Broadcast.BroadcastFunction(!)"
 @test eval(:(.+)) == Base.BroadcastFunction(+)
 
+@testset "Issue #5187: Broadcasting of short-circuiting ops" begin
+    ex = Meta.parse("A .< 1 .|| A .> 2")
+    @test ex == :((A .< 1) .|| (A .> 2))
+    @test ex.head == :call
+    @test ex.args[1] == Symbol(".||")
+    ex = Meta.parse("A .< 1 .&& A .> 2")
+    @test ex == :((A .< 1) .&& (A .> 2))
+    @test ex.head == :call
+    @test ex.args[1] == Symbol(".&&")
+
+    A = -1:4
+    @test (A .< 1 .|| A .> 2) == [true, true, false, false, true, true]
+    @test (A .>= 1 .&& A .<= 2) == [false, false, true, true, false, false]
+
+    mutable struct F5187; x; end
+    (f::F5187)(x) = (f.x += x)
+    @test (iseven.(1:4) .&& (F5187(0)).(ones(4))) == [false, 1, false, 2]
+    @test (iseven.(1:4) .|| (F5187(0)).(ones(4))) == [1, true, 2, true]
+    @test (.&&)(iseven.(1:4), (F5187(0)).(ones(4))) == [false, 1, false, 2]
```

Comment:
Do we actually want to allow `(.&&)(a, b)`? We don't allow that for `&&`.

Reply:
If `f = (.&&)` by itself forms a callable function, it seems a little crazy to disallow `(.&&)(a,b)`.

Reply:
This came up when trying to make `.&&` and `.||` syntactic operators instead, i.e. giving them their own expression heads, since we don't currently allow this syntax there. The more I think about this though, the more I think we should just follow the semantics of `&&` and `||` here and only allow them as infix operators, disallowing them as standalone functions as well. I think making them behave more like regular functions is just unnecessarily confusing, since they are really not and I can't come up with any real world use case where you'd actually need standalone `.&&` over `.&`.

Reply:
> If f = (.&&) by itself forms a callable function, it seems a little crazy to disallow (.&&)(a,b).

Sounds like that should be disallowed too.

Reply:
@mbauman If you are ok with that I can push my changes to this branch, otherwise I will hold off for now.

Reply:
Yeah, we could remove that part of it.  It's the conservative behavior.  If folks want it, it can always be added later.